File: src/main/java/org/library/LibraryAuth.java
package org.library;

public class LibraryAuth {
    private Borrowers users;

    public LibraryAuth(Borrowers borrowers){
        users = borrowers;
    }

    public AuthEnum authUser(String username, String password){
        if (!validateUsername(username) || !validatePassword(password)){
            return AuthEnum.INVALID_INPUT;
        }

        User user;
        try {
            user = users.getBorrower(username);

            if (!user.matchPassword(password))
                return AuthEnum.INVALID_CREDENTIALS;

        } catch (RuntimeException e) {
            return AuthEnum.INVALID_CREDENTIALS;
        }

        return AuthEnum.SUCCESS;
    }

    private boolean validateUsername(String u){
        return !(u.isEmpty());
    }
    private boolean validatePassword(String p){
        return !(p.isEmpty());
    }
}

File: main/LibraryAuth.js
import { AuthEnum } from "./AuthEnum.js";

export default class LibraryAuth {
  constructor(borrowers) {
    this.users = borrowers;
  }

  authUser(username, password) {
    if (!this.validateUsername(username) || !this.validatePassword(password)) {
      return AuthEnum.INVALID_INPUT;
    }

    let user;
    try {
      user = this.users.getBorrower(username);

      if (
        !user ||
        typeof user.matchPassword !== "function" ||
        !user.matchPassword(password)
      )
        return AuthEnum.INVALID_CREDENTIALS;
    } catch (e) {
      return AuthEnum.INVALID_CREDENTIALS;
    }

    return AuthEnum.SUCCESS;
  }

  validateUsername(u) {
    return typeof u === "string" && u.length > 0;
  }

  validatePassword(p) {
    return typeof p === "string" && p.length > 0;
  }
}
